### ISSUE TITLE
01-logistic-regression-bank-marketing-campaign-data.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/benjaminmullermiranda-a11y-Machine_Learning#1]
### **1. Estructura Lógica Clara**
```
1. Cargar datos
2. Explorar (shape, dtypes, nulos)
3. Limpiar (drop duration)
4. Codificar (LabelEncoder)
5. Dividir (train/test)
6. Modelo base
7. Optimización (Pipeline + GridSearchCV)
8. Evaluación
```

### **2. Decisión: Eliminar `duration`**
Eliminas `duration` porque:
- Es información que **solo se conoce después** de la llamada
- Bien explicado y decisión tomada pero, ¿puede ser que importe debido a que la duración si que te dice que cuanto más dura más se contrata?
- Es solo una puesta en vista, sin embargo tu decisión está bien. 

### **3. Uso Correcto de Pipeline + GridSearchCV**

Usas:
-  Pipeline (scaler + model juntos)
-  Buscas en C, solver, y class_weight
-  Cross-validation (cv=5)
-  Métrica f1 (apropiada para datos desbalanceados)

### **4. Análisis de Trade-offs**
- **Accuracy 90% vs 81%** → El 90% es engañoso (predice "No" siempre)
- **Recall 20% vs 65%** → Mejora 3x en detectar compradores reales
- **False Positives 71 → 1,217** → El costo está justificado

**Esto es pensamiento crítico.** 

---

## **Áreas Para Mejorar**

Una lista de cosas que podrías darle una vuelta o mejorar. No necesariamente erróneas pero quiero que chequees, SOBRETODO TEMA EDA:

Cuidado con esta línea, si ejecutas en orden como se suele hacer te da fallo ya que df aún no está definido.
Ponlo después de llamar al dataset de la url.

<img width="922" height="636" alt="Image" src="https://github.com/user-attachments/assets/b33bb82b-d52f-4469-9f22-df2ecbbb9043" />

---

### **1. Encoding: LabelEncoder en Lugar de One-Hot**

**Lo que haces:**
```python
le = LabelEncoder()
for col in cat_cols:
    df_model[col] = le.fit_transform(df_model[col])
```

**Problema:**
- LabelEncoder asigna 0, 1, 2, 3... a categorías
- Esto **implica un orden** que no existe
- Ej: `job` → {0: housemaid, 1: services, 2: admin...}
- El modelo piensa que `job=2` es "mayor" que `job=1`

**Mejor Alternativa:**
```python
# One-Hot Encoding
df_model = pd.get_dummies(df_model, drop_first=True, columns=cat_cols)
```

**Impacto:** Puede afectar la calidad del modelo, aunque en LR muchas veces funciona similar.

---

### **2. No Hay Análisis Exploratorio (EDA) Real**

```python
print(df.shape)           # 41,188 filas
print(df['y'].value_counts())  # Desbalance 88%/12%
```

**Te falta:**
- Correlaciones entre features y target
- Distribuciones de variables (histogramas)
- Valores atípicos (outliers)
- Análisis bivariado (age vs target, campaign vs target)
- Entender qué variables son realmente útiles

Saltas directamente al modelado casi. 

No sabes si tus features son buenos realmente y te falta una parte importante del proyecto.
